core/src/main/java/org/commonjava/indy/core/content/DefaultDirectContentAccess.java

```diff
@@ -111,4 +158,56 @@ public boolean exists( ArtifactStore store, String path )
         return downloadManager.list( store, path );
     }
 
+    @Override
+    public Map<String, List<StoreResource>> listRaw( ArtifactStore store,
+                                                     List<String> parentPathList ) throws IndyWorkflowException
+    {
+        ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
I think the same comments from above apply to this method too...to preserve the order of paths in the original request, use @WeftManaged / @ExecutorConfig to inject the executor service, and possibly to use ExecutorCompletionService.